File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $seen = [];
        
        foreach($nums as $key => $num){
            $t = $target - $num;
            
            if(isset($seen[$t])){
                return [$seen[$t], $key];
            }
            
            $seen[$num] = $key;
        }
    }
}

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int, len(nums))
    
    for key, val := range nums {
        if needed, ok := seen[target - val]; ok{
            return []int{needed, key}
        }
        seen[val] = key
    }
    
    return []int{0, 0}
}